### PR TITLE
Feature/topology visualizer contract #205

### PR DESCRIPTION
**Pull request by Flyur**
API contract for the stream topology visualization feature.
Please, review the data model for the API.

**How Has This Been Tested?** (put an "X" next to an item)

- [ ] No need to
- [X] Manually(please, describe, when necessary)
- [ ] Unit checks
- [ ] Integration checks
- [X] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [X] My changes generate no new warnings(e.g. Sonar is happy)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 1]
```diff
@@ -2098,3 +2126,99 @@ components:
...
         - topicName
+
+    TopologyNode:
+      type: object
+      properties:
+        name:
+          type: string
+        type:
+          type: string
+          enum:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:2137]: Please move enum to separate class

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 2]
```diff
@@ -1251,6 +1251,34 @@ paths:
...
+      operationId: getStreamTopology
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GlobalTopology'
+
+  /api/streamtopology/dummy:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:1268]: Let's skip dummy endpoints in source code

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 3]
```diff
@@ -1251,6 +1251,34 @@ paths:
         404:
           description: Not found
 
+  /api/streamtopology:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:1254]: We should do it on cluster level and per application

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml (hunk 2 above)
Comment (Flyur) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml]: done

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml (hunk 3 above)
Comment (Flyur) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml]: done

**Comment by Flyur** [comment]
Simplified the data model for now - it contains information about the topology of only one application and without kafka connectors.
simple example of the json:

`{
    "processorsNumber": null,
    "topicsNumber": null,
    "topology": {
        "adjacency": {
            "sub-topology 0": [
                "count-resolved-repartition"
            ],
            "count-resolved-repartition": [
                "sub-topology 1"
            ],
            "sub-topology 1": []
        },
        "nodes": {
            "sub-topology 0": {
                "name": "sub-topology 0",
                "type": "SUB_TOPOLOGY",
                "subTopology": {
                    "adjacency": {
                        "KSTREAM-SOURCE-0000000000": [
                            "KSTREAM-TRANSFORM-0000000001"
                        ],
                        "KSTREAM-TRANSFORM-0000000001": [
                            "KSTREAM-SINK-000000002"
                        ],
                        "KSTREAM-SINK-000000002": []
                    },
                    "nodes": {
                        "KSTREAM-SOURCE-0000000000": {
                            "name": "KSTREAM-SOURCE-0000000000",
                            "type": "SOURCE_PROCESSOR",
                            "topics": [
                                "inputTopic"
                            ]
                        },
                        "KSTREAM-TRANSFORM-0000000001": {
                            "name": "KSTREAM-TRANSFORM-0000000001",
                            "type": "PROCESSOR",
                            "stores": [
                                "store 1"
                            ]
                        },
                        "KSTREAM-SINK-000000002": {
                            "name": "KSTREAM-SINK-000000002",
                            "type": "SINK_PROCESSOR",
                            "topic": "count-resolved-repartition"
                        }
                    }
                }
            },
            "count-resolved-repartition": {
                "name": "count-resolved-repartition",
                "type": "TOPIC",
                "inputMessagesNumber": null,
                "outputMessagesNumber": null,
                "lag": null
            },
            "sub-topology 1": {
                "name": "sub-topology 1",
                "type": "SUB_TOPOLOGY",
                "subTopology": {
                    "adjacency": {
                        "KSTREAM-SOURCE-000000003": [
                            "KSTREAM-TRANSFORM-000000004"
                        ],
                        "KSTREAM-TRANSFORM-000000004": [
                            "KSTREAM-SINK-000000005"
                        ],
                        "KSTREAM-SINK-000000005": []
                    },
                    "nodes": {
                        "KSTREAM-SOURCE-000000003": {
                            "name": "KSTREAM-SOURCE-000000003",
                            "type": "SOURCE_PROCESSOR",
                            "topics": [
                                "count-resolved-repartition"
                            ]
                        },
                        "KSTREAM-TRANSFORM-000000004": {
                            "name": "KSTREAM-TRANSFORM-000000004",
                            "type": "PROCESSOR",
                            "stores": [
                                "store 2"
                            ]
                        },
                        "KSTREAM-SINK-000000005": {
                            "name": "KSTREAM-SINK-000000005",
                            "type": "SINK_PROCESSOR",
                            "topic": "streams-count-resolved"
                        }
                    }
                }
            }
        }
    }
}`

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml (hunk 1 above)
Comment (Flyur) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml]: done

**Comment by germanosin** [comment]
LGTM for contract, let's implement reading / parsing part

**Comment by Flyur** [comment]
@germanosin should we merge it to unblock frontend?

**Pull request closed without merging**